viadot/flows/transform_and_catalog.py

```diff
@@ -0,0 +1,254 @@
+            token=self.token,
+            token_secret=self.token_secret,
+            path=local_dbt_repo_path,
+            flow=self,
+        )
+
+        # dbt CLI does not handle passing --target=None
+        dbt_target_option = (
+            f"-t {self.dbt_target}" if self.dbt_target is not None else ""
+        )
+
+        if self.stateful:
+            source_freshness_upstream = clone
+        else:
+            # Clean up artifacts from previous runs (`target/` dir and packages)
+
+            dbt_clean_up = ShellTask(
+                name="dbt_task_clean",
+                command=f"dbt clean",
+                helper_script=f"cd {self.dbt_project_path}",
+                return_all=True,
+                stream_output=True,
+            ).bind(flow=self)
+
+            pull_dbt_deps = ShellTask(
+                name="dbt_task_deps",
+                command=f"dbt deps",
+                helper_script=f"cd {self.dbt_project_path}",
+                return_all=True,
+                stream_output=True,
+            ).bind(flow=self)
+            # source_freshness_upstream = pull_dbt_deps
+
+        # Source freshness
+        # Produces `sources.json`
+        # source_freshness_select = self.dbt_selects.get("source_freshness")
+        # source_freshness_select_safe = (
+        #     f"-s {source_freshness_select}"
+        #     if source_freshness_select is not None
+        #     else ""
+        # )
+
+        # source_freshness = DBTTask(name="dbt_task_source_freshness").bind(
+        #     project_path=self.dbt_project_path,
+        #     command=f"source freshness {source_freshness_select_safe} {dbt_target_option}",
+        #     flow=self,
+        # )
+        # source_freshness = ShellTask(
+        #     name="dbt_task_source_freshness",
+        #     command=f"dbt source freshness {source_freshness_select_safe} {dbt_target_option}",
+        #     helper_script=f"cd {self.dbt_project_path}",
+        #     return_all=True,
+        #     stream_output=True,
+        # ).bind(flow=self)
+
+        run_select = self.dbt_selects.get("run")
+        run_select_safe = f"-s {run_select}" if run_select is not None else ""
+
+        run = ShellTask(
+            name="dbt_task_run",
+            command=f"dbt run {run_select_safe} {dbt_target_option}",
+            helper_script=f"cd {self.dbt_project_path}",
+            return_all=True,
+            stream_output=True,
+        ).bind(flow=self)
+
+        # Generate docs
+        # Produces `catalog.json`, `run-results.json`, and `manifest.json`
+        generate_catalog_json = ShellTask(
+            name="dbt_task_docs_generate",
+            command=f"dbt docs generate {dbt_target_option} --no-compile",
+            helper_script=f"cd {self.dbt_project_path}",
+            return_all=True,
+            stream_output=True,
+        ).bind(flow=self)
+
+        test_select = self.dbt_selects.get("test", run_select)
```

Comment:
maybe change the name from "test" to sth different, test suggests that it's something just fore testing not production

Reply:
It was migrated from `2.0`. I will ask how it can be changed.